.drone.yml

```diff
@@ -1,129 +1,119 @@
 pipeline:
   test_build:
-    image: docker:1.11
+    image: docker:17.09.0-ce
     environment:
-      - DOCKER_HOST=tcp://127.0.0.1:2375
+      - DOCKER_HOST=tcp://172.17.0.1:2375
     commands:
-      - apk add --no-cache bash sed grep bc coreutils
+      - apk add --no-cache bash sed grep bc coreutils curl
       - ./ci-build.sh
     when:
-      event: [push, pull_request, tag]
+      event: [push, pull_request]
   
   build_clamav:
-    image: docker:1.11
+    image: docker:17.09.0-ce
     environment:
-      - DOCKER_HOST=tcp://127.0.0.1:2375
+      - DOCKER_HOST=tcp://172.17.0.1:2375
     commands:
-      - docker build -t clamav .
+      - docker build -t clamav:$${DRONE_COMMIT_SHA} .
     when:
-      branch: [master, refs/tags/*]
-      event: [push, tag]
+      branch: master
+      event: push
   
   build_clamav-rest:
-    image: docker:1.11
-    environment:
-      - DOCKER_HOST=tcp://127.0.0.1:2375
-    commands:
-      - docker build -t clamav-rest clamav-rest
-    when:
-      branch: [master, refs/tags/*]
-      event: [push, tag]
-  
-  clamav_to_quay:
-    image: docker:1.11
-    environment:
-      - DOCKER_HOST=tcp://127.0.0.1:2375
-    commands:
-      - docker login -u="ukhomeofficedigital+drone" -p=${CLAMAV_REPO_TOKEN} quay.io
-      - docker tag clamav quay.io/ukhomeofficedigital/clamav:${DRONE_COMMIT_SHA}
-      - docker push quay.io/ukhomeofficedigital/clamav:${DRONE_COMMIT_SHA}
-    when:
-      branch: [master, refs/tags/*]
-      event: [push, tag]
-  
-  clamav-rest_to_quay:
-    image: docker:1.11
-    environment:
-      - DOCKER_HOST=tcp://127.0.0.1:2375
-    commands:
-      - docker login -u="ukhomeofficedigital+drone" -p=${CLAMAV_REPO_TOKEN} quay.io
-      - docker tag clamav-rest quay.io/ukhomeofficedigital/clamav-rest:${DRONE_COMMIT_SHA}
-      - docker push quay.io/ukhomeofficedigital/clamav-rest:${DRONE_COMMIT_SHA}
-    when:
-      branch: [master, refs/tags/*]
-      event: [push, tag]
-  
-  trigger_deploy_to_platform-services_dev:
-    image: quay.io/ukhomeofficedigital/kd:v0.2.2
+    image: docker:17.09.0-ce
     environment:
-      - DOCKER_HOST=tcp://127.0.0.1:2375
+      - DOCKER_HOST=tcp://172.17.0.1:2375
     commands:
-      - kd --insecure-skip-tls-verify --namespace=virus-scan --kube-server=https://kube-dev.dsp.notprod.homeoffice.gov.uk --retries 50 --kube-token=${KUBE_TOKEN_DEV} --file k8s/clamd-api.yaml
+      - docker build -t clamav-rest:$${DRONE_COMMIT_SHA} clamav-rest
     when:
       branch: master
       event: push
```

Comment:
Similarly here. Could use `event: [ push, tag ]` instead of having separate step below.

Reply:
I left a comment on your other comment from yesterday:

> You can but then you wouldn't be able to specify the master branch (for the push event). So either you have 2 steps each (as it is now) or remove the branch condition so that it triggers the pipeline on every push on every branch (as well as triggering the test build pipeline). I felt that this way was better.

Reply:
Hmm, does the use of multiple event types invalidate branch selector?

Reply:
It would seem so. I did try it the way you're suggesting it but the pipeline didn't trigger.

Reply:
were you tagging master or some other branch?

Reply:
ooooooooooooooooooooooh. That was probably why it didn't work. Whoops. I'll change it to `event: [ push, tag ]`. Although I obviously won't be able to test that the pipelines work when tagging until this is merged into master.

Reply:
that's OK.